### ISSUE TITLE
About adding new widgets

### ISSUE THREAD

**Issue by spcan** [xi-editor/druid#589]
Currently I'm working on a project of my own and I've ended developing some widgets for this project.

Is there an interest of adding more widgets to the Druid project currently? If so how, is there a process to add them or just opening a PR is enough?

**Comment by cmyr** [comment]
It depends on the nature of the widget; we'd be happy to consider standard control-type widgets, but more exotic things might belong in separate crates. (We should have a section in the docs or readme to point to external druid widgets & utilities).

What are the widgets you've been working on?

**Comment by spcan** [comment]
The ones I've worked most on are a kind of range marker (basically, show with rectangles and marks a value inside or outside a given range), a widget that mixes two images together and a tabs like widget.

Except the tabs one, pretty much everything else falls under the exotic type of widget.

**Comment by cmyr** [comment]
cool. We're definitely interested in a tabs widget; there was initial work in this direction in #507. @s3thi had mentioned wanting to build a tab widget on top of that, but I'm not sure if there has been any progress there; in any case you would be welcome to PR something.

The place where I could imagine the other things being interesting would be in `examples`; I would like there to be some more full-featured things there, and some custom widgets might make sense?

**Comment by s3thi** [comment]
I haven't been working on a tab switcher -- mostly because my app doesn't need one -- so that task is open for whoever wants to pick it up.

I've actually been using the `ViewSwitcher` to build a sidebar widget, which I find to be more appropriate for my app. It looks something like the view you see on this page: https://stackoverflow.com/questions/58785044/swiftui-sidebar-list-does-not-register-first-click. I would imagine that a sidebar and a tab switcher mostly differ in the painting logic and could be built on top of the same `ViewSwitcher` code.

**Comment by spcan** [comment]
Yeah, I don't see much difference between the logic needed to change from the sidebar to the tabs. Seeing this, I don't know if your widget and mine would overlap a bit, so maybe if yours comes out first mine is not needed anymore.
I wasn't planning to use the `ViewSwitcher` to change views, but I'll look into it now.

**Comment by s3thi** [comment]
@spcan I think you should go ahead with the tab widget. My sidebar has enough app-specific functionality at this point that it wouldn't make sense for Druid at all.
